src/ui/public/vis/editors/default/update_editor_state.js

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { cloneDeep } from 'lodash';
+
+export const updateEditorStateWithChanges = (newState, oldState, editorState) => {
+  let dirty = false;
+  for (const prop in newState) {
+    if (newState.hasOwnProperty(prop)) {
+      const newStateValue = newState[prop];
+      const oldStateValue = oldState[prop];
+      const editorStateValue = editorState[prop];
+
+      if (newStateValue && typeof newStateValue === 'object') {
+        if (editorStateValue) {
+          // Keep traversing.
+          if (updateEditorStateWithChanges(newStateValue, oldStateValue, editorStateValue)) {
+            dirty = true;
```

Comment:
Any reason this and other checks can't just `return true` to short circuit all other execution? I hope that checking for changes doesn't have some other effect on things...

Reply:
the null values were treated incorectly (that was the problem)
we cant just return as we are udatingEditorState not checking if it changed ....